### PR TITLE
document mq manager

### PR DESCRIPTION
**Pull request by CaroMac**

### CODE REVIEW

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+**Alpha**
+
+## Overview
+This Manager provides the ability to connect to an existing Queue Manager, enabling applications to read and write one or more messages to and from a queue. <br><br> 
+
+
+## Annotations
+
+The following annotations are available with the CICS TS CECI Manager
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:14]: MQ manager

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+<details>
+<summary>MQ Manager</summary>
+
+| Annotation: |MQ Manager |
+| --------------------------------------- | :------------------------------------- |
+| Name: | @QueueManager |
+| Description: | The <code>@QueueManager</code> annotation represents the name of the MQ Queue Manager |
+| Syntax: | @QueueManager<br> public IMessageQueueManager qmgr;<br> |
+| Notes: | The <code>IMessageQueueManager</code> interface enables connection to the MQ Queue Manager.  |
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:23]: We should detail the attributes available

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+<details>
+<summary>MQ Queue</summary>
+
+| Annotation: |MQ Queue |
+| --------------------------------------- | :------------------------------------- |
+| Name: | @Queue |
+| Description: | The <code>@Queue</code> annotation represents the name of the MQ Queue |
+| Syntax: | @Queue<br> public IMessageQueue queue;<br> |
+| Notes: | The <code>IMessageQueue</code> interface enables the test to put the provided messages onto the queue and retrieve messages from the queue. |
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:34]: We should detail the attributes available

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
+---
+path: "/docs/managers/mq-manager"
+title: "MQ Manager"
+---
+
+**Alpha**
+
+## Overview
+This Manager provides the ability to connect to an existing Queue Manager, enabling applications to read and write one or more messages to and from a queue. <br><br>
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:9]: Need to explain it's for a IBM MQ queue manager

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+| Notes: | The <code>IMessageQueue</code> interface enables the test to put the provided messages onto the queue and retrieve messages from the queue. |
+</details>
+
+
+
+## Code snippets
+
+Use the following code snippets to help you get started with the MQ Manager.
+ 
+<details><summary>Create a MQ Manager</summary>
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:43]: We should be clear we do not provision a new queue manager, or queues. But connect the test to existing queue managers and queues

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+</details>
+
+<details><summary>Instantiate a MQ Queue</summary>
+
+The following snippet shows the code that is required to instantiate a MQ Queue:
+
+```java
+@Queue(archive = "true", name = "GALASA.INPUT.QUEUE")
+public IMessageQueue queue;
+```
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:60]: Need to show the tag attribute that shows which qmgr the queue is on

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+</details>
+
+<details><summary>Read messages from a queue</summary>
+
+The following snippet shows the code required to read a message from a queue:
+
+```java
+Message m = queue.getMessage();
+String response = m.getBody(String.class);
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:92]: I need to check this code example as if this is the current api I can make that better

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+```
+</details>
+
+<details><summary>Clear messages from a queue</summary>
+
+The following snippet shows the code required to clear messages from a queue:
+
+```java
+queue.clearQueue();
+Message m = queue.getMessageNoWait();
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:102]: We don't need this line

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+```
+</details>
+
+## Configuration Properties
+
+The following are properties used to configure the MQ Manager.
+ 
+<details>
+<summary>Queue Manager channel CPS Property</summary>
+
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:112]: I think we are missing the attribute that connects the tag to the instance

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+</details>
+
+<details><summary>Put messages to a queue</summary>
+
+The following snippet shows the code required to put a message on a queue:
+
+```java
+TextMessage tm = qmgr.createTextMessage(testData);
+queue.sendMessage(tm);
+```
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:83]: Could we show the methods to create a text or binary message

**Review by hobbit1983 — CHANGES_REQUESTED** [review]
That was quick!  Nice work @CaroMac just some tweaks and updates

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 1 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:14]: updated

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 4 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:9]: Have added in extra content

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 5 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:43]: updated

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 8 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:102]: removed

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 10 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:83]: added

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 2 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:23]: added

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 3 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:34]: added

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 9 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:112]: Should this be in this file? https://github.com/galasa-dev/managers/blob/iss835/galasa-managers-parent/galasa-managers-comms-parent/dev.galasa.mq.manager/src/main/java/dev/galasa/mq/internal/properties/InstanceChannelName.java

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 6 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:60]: updated

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 9 above)
Comment (hobbit1983) [reply src/markdown-pages/docs/managers/mq-manager.md:136]: Yes - some of the documentation on the properties was incorrect.  That has been updated the and the PR rectified.  I have also added a new CPS property to get the credentials

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 7 above)
Comment (hobbit1983) [reply src/markdown-pages/docs/managers/mq-manager.md:117]: no that is correct and will stay like that - sorry

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 11]
```diff
@@ -0,0 +1,188 @@
+---
+path: "/docs/managers/mq-manager"
+title: "MQ Manager"
+---
+
+**Alpha**
+
+## Overview
+This Manager provides the ability to connect a test to an existing IBM MQ queue manager, enabling one or more messages to be written to and read from an existing IBM MQ queue. <br><br>
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:9]: From existing queues

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+
+| Annotation: |MQ Manager |
+| --------------------------------------- | :------------------------------------- |
+| Name: | @QueueManager |
+| Description: | The <code>@QueueManager</code> annotation represents the name of the IBM MQ queue manager |
+| Attribute: `name` |  The name of the IBM MQ queue manager |
+| Attribute: `host` |  The host name of the system on which the IBM MQ queue manager is running |
+| Attribute: `port` |  The port number to connect to the IBM MQ queue manager |
+| Attribute: `channel` |  The name of the channel used to move messages  |
+| Attribute: `manager` |  The Galasa implementation of the IBM MQ queue manager |
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:26]: these are not attributes of the annotation - but CPS properties

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 11 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md]: updated

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 12 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md]: added annotation attributes

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 9 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:132]: updated docs as per changes to file

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 13]
```diff
@@ -0,0 +1,227 @@
...
+
+The following annotations are available with the MQ Manager
+<details>
+<summary>MQ Manager</summary>
+
+| Annotation: |MQ Manager |
+| --------------------------------------- | :------------------------------------- |
+| Name: | @QueueManager |
+| Description: | The <code>@QueueManager</code> annotation represents the name of the IBM MQ queue manager |
+| Attribute: `queueMgrTag` |  Specifies which queue manager to connect to. Default value is _PRIMARY_. |
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:21]: This has been renamed to just tag

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 14]
```diff
@@ -0,0 +1,227 @@
...
+<details>
+<summary>MQ Queue</summary>
+
+| Annotation: |MQ Queue |
+| --------------------------------------- | :------------------------------------- |
+| Name: | @Queue |
+| Description: | The <code>@Queue</code> annotation represents the name of the IBM MQ queue |
+| Attribute: `queueMgrTag` |  Specifies which queue manager to connect to. Default value is _PRIMARY_.  |
+| Attribute: `name` |  Specifies the name of the queue as it appears on the queue manager. Use the `name` attribute if the queue name never changes, regardless of environment.  |
+| Attribute: `tag` |  The name of the queue. Use the `tag` attribute when the value of the queue name is defined by a property in the CPS file.  |
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:35]: just say CPS, remove the file

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 15]
```diff
@@ -0,0 +1,227 @@
...
+| Notes: | You must specify either the `name` or the `tag` attribute but not both. If you specify both or neither, an exception is generated. |
+| Attribute: `archive` |  Store log data sets in the RAS to aid debugging. Default value is _true_. Valid values are _true_ and _false_. |
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:37]: might be better to say 'archive retrieved and sent messages to / from this queue into the the RAS...`

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 16]
```diff
@@ -0,0 +1,227 @@
...
+| Default value: | None |
+| Valid values: |  |
+| Examples: | <code>mq.server.[instanceid].port=1414</code> |
+
+</details>
+
+<details>
+<summary>Instance ID for the tag CPS Property</summary>
+
+| Property: | Instance ID for the tag CPS Property |
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:203]: This should be the first property mentioned and we should show how it relates to the tag attributes in the @QueueManager annotation

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 17]
```diff
@@ -0,0 +1,227 @@
...
+| Default value: | None |
+| Valid values: |  |
+| Examples: | <code>mq.tag.[tag].instanceid=QUEUEMGR1</code> |
+
+</details>
+
+<details>
+<summary>Queue name for the tag CPS Property</summary>
+
+| Property: | Queue name for the tag CPS Property |
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:217]: Can we also show how this relates to the tag attribute in the @queue annotation

**Review by hobbit1983 — CHANGES_REQUESTED** [review]

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 17 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md]: updated

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 16 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md]: updated

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 18]
```diff
@@ -0,0 +1,227 @@
...
+
+The following snippets show the code required to create a text or binary message to put on an IBM MQ queue:
+
+```java
+TextMessage tm = qmgr.createTextMessage(testData);
+queue.sendMessage(tm);
+```
+```java
+TextMessage tm = qmgr.createTextMessage(testData);
+queue.sendMessage(tm,tm,tm,tm,tm,tm,tm,tm);
```
[the remark this answers is not in the document]
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:97]: added

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 15 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md]: done

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 14 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md]: removed

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 13 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md]: updated

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 19]
```diff
@@ -0,0 +1,233 @@
...
+
+<details><summary>Instantiate an IBM MQ queue</summary>
+
+The following snippet shows the code that is required to instantiate an IBM MQ queue:
+
+```java
+@QueueManager()
+public IMessageQueueManager qmgr;
+
+@Queue(archive = "true", name = "GALASA.INPUT.QUEUE")
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:66]: need to remove the quotes around the value true.  It was a String , but I've changed it to a boolean so the quotes are no longer needed

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 20]
```diff
@@ -0,0 +1,233 @@
...
+public IMessageQueue queue;
+```
+
+You can just as simply instantiate multiple IBM MQ queues:
+
+```java
+@QueueManager()
+public IMessageQueueManager qmgr;
+
+@Queue(archive = "true", name = "GALASA.INPUT.QUEUE")
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:76]: remove the quotes around true

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 21]
```diff
@@ -0,0 +1,233 @@
...
+public IMessageQueue queue;
+	
+@Queue(archive = "false", name = "GALASA.INPUT.QUEUE2")
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:79]: remove the quotes around false

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 22]
```diff
@@ -0,0 +1,233 @@
...
+public IMessageQueue queue2;
+	
+@Queue(tag = "NEWQUEUE")
+public IMessageQueue queue3;
+```
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:84]: should we expand the example (or have two to show multiple queues on multiple queue managers to show the rest of the attributes?

File: src/markdown-pages/docs/managers/mq-manager.md [hunk 23]
```diff
@@ -0,0 +1,233 @@
...
+<summary>Queue name for the tag CPS Property</summary>
+
+| Property: | Queue name for the tag CPS Property |
+| --------------------------------------- | :------------------------------------- |
+| Name: | mq.server.[tag].queuename |
+| Description: | The queue name for the specified tag |
+| Required:  | Yes |
+| Default value: | None |
+| Valid values: |  |
+| Examples: | <code>mq.server.[tag].queuename=GALASA.INPUT.QUEUE</code> |
```
Comment (hobbit1983) [inline src/markdown-pages/docs/managers/mq-manager.md:156]: is actually mq.queue.tag.<tag>.queuename

**Review by hobbit1983 — CHANGES_REQUESTED** [review]

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 19 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:66]: removed

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 20 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:76]: removed

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 21 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:79]: removed

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 23 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:156]: thanks, updated

File: src/markdown-pages/docs/managers/mq-manager.md (hunk 22 above)
Comment (CaroMac) [reply src/markdown-pages/docs/managers/mq-manager.md:84]: added an extra example

**Review by hobbit1983 — APPROVED** [review]

**Pull request merged**